pkg/downloader/chart_downloader.go

```diff
@@ -133,6 +137,26 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 	return destfile, ver, nil
 }
 
+func (c *ChartDownloader) DownloadToInMemoryCache(ref, version string) (string, *provenance.Verification, error) {
```

Comment:
@waveywaves any specific reason we are returning `*provenance.Verification`  ? since anyways we return `nil` in all paths.

Reply:
I have updated to code to return a non null, provenance verification is important to check the source of the chart against the given keyring. I would like to cover this in another PR if that is fine as it would involve some refactoring as well.